plugins/TwoFactorAuth/TwoFactorAuth.php

```diff
@@ -265,7 +265,7 @@ private function requiresAuth($module, $action, $parameters)
         }
 
         $auth = StaticContainer::get('Piwik\Auth');
-        if ($auth && !$auth->getLogin() && method_exists($auth, 'getTokenAuth') && $auth->getTokenAuth()) {
```

Comment:
If `Piwik\Auth` object can't be created an exception should be thrown, so checking for its existence should be unneeded.